hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/MultiDCAwareService.java

```diff
@@ -62,36 +65,67 @@ public String readMessageFromPrimary(
         .readMessageFromPrimary(topic, partition, offset);
   }
 
-  public MultiDCOffsetChangeSummary retransmit(
-      Topic topic, String subscriptionName, Long timestamp, boolean dryRun, RequestUser requester) {
+  public MultiDCOffsetChangeSummary fetchTopicOffsetsAt(Topic topic, Long timestamp) {
     MultiDCOffsetChangeSummary multiDCOffsetChangeSummary = new MultiDCOffsetChangeSummary();
 
     clusters.forEach(
         cluster ->
             multiDCOffsetChangeSummary.addPartitionOffsetList(
-                cluster.getClusterName(),
-                cluster.indicateOffsetChange(topic, subscriptionName, timestamp, dryRun)));
-
-    if (!dryRun) {
-      logger.info(
-          "Starting retransmission for subscription {}. Requested by {}. Retransmission timestamp: {}",
-          topic.getQualifiedName() + "$" + subscriptionName,
-          requester.getUsername(),
-          timestamp);
-      multiDcExecutor.executeByUser(
-          new RetransmitCommand(new SubscriptionName(subscriptionName, topic.getName())),
-          requester);
-      clusters.forEach(clusters -> waitUntilOffsetsAreMoved(topic, subscriptionName));
-      logger.info(
-          "Successfully moved offsets for retransmission of subscription {}. Requested by user: {}. Retransmission timestamp: {}",
-          topic.getQualifiedName() + "$" + subscriptionName,
-          requester.getUsername(),
-          timestamp);
-    }
+                cluster.getClusterName(), cluster.fetchTopicOffsetsAt(topic, timestamp)));
 
     return multiDCOffsetChangeSummary;
   }
 
+  public void moveOffsets(
+      Topic topic,
+      String subscriptionName,
+      Map<String, List<PartitionOffset>> brokerPartitionOffsets) {
+    clusters.forEach(
+        cluster ->
+            cluster.validateIfOffsetsCanBeMoved(
+                topic, new SubscriptionName(subscriptionName, topic.getName())));
+
+    clusters.forEach(
+        cluster ->
+            cluster.moveOffsets(
+                new SubscriptionName(subscriptionName, topic.getName()),
+                brokerPartitionOffsets.getOrDefault(
+                    cluster.getClusterName(), Collections.emptyList())));
+  }
+
+  public void moveOffsetsForActiveConsumers(
+      Topic topic,
+      String subscriptionName,
+      Map<String, List<PartitionOffset>> brokerPartitionOffsets,
+      RequestUser requester) {
+    clusters.forEach(
+        cluster ->
+            cluster.validateIfOffsetsCanBeMovedByConsumers(
+                topic, new SubscriptionName(subscriptionName, topic.getName())));
+
+    clusters.forEach(
+        cluster ->
+            Optional.ofNullable(brokerPartitionOffsets.get(cluster.getClusterName()))
+                .ifPresent(
+                    offsets -> cluster.indicateOffsetChange(topic, subscriptionName, offsets)));
+
+    logger.info(
+        "Starting offsets move for subscription {}. Requested by {}. Retransmission offsets: {}",
```

Comment:
minor: maybe "Starting moving offsets" would sound better? let me know what do You think about it